src/Xamarin.Android.Build.Tasks/Utilities/MavenExtensions.cs

```diff
@@ -0,0 +1,229 @@
+			using var src = await artifact.OpenLibraryFile (artifact.Versions.First (), Path.GetExtension (filename));
+			using var sw = File.Create (filename);
+
+			await src.CopyToAsync (sw);
+
+			return null;
+		} catch (Exception ex) {
+			return ex.Message;
+		}
+	}
+
+	public static string GetRepositoryCacheName (this Artifact artifact)
+	{
+		var type = artifact.Repository;
+
+		if (type is MavenCentralRepository)
+			return "central";
+
+		if (type is GoogleMavenRepository)
+			return "google";
+
+		if (type is UrlMavenRepository url) {
+			using var hasher = SHA256.Create ();
+			var hash = hasher.ComputeHash (Encoding.UTF8.GetBytes (url.BaseUri.ToString ()));
+			return Convert.ToBase64String (hash);
+		}
+
+		// Should never be hit
+		throw new ArgumentException ($"Unexpected repository type: {type.GetType ()}");
+	}
+
+	public static void FixDependency (Project project, Project? parent, Dependency dependency)
+	{
+		// Handle Parent POM
+		if ((string.IsNullOrEmpty (dependency.Version) || string.IsNullOrEmpty (dependency.Scope)) && parent != null) {
+			var parent_dependency = parent.FindParentDependency (dependency);
+
+			// Try to fish a version out of the parent POM
+			if (string.IsNullOrEmpty (dependency.Version))
+				dependency.Version = ReplaceVersionProperties (parent, parent_dependency?.Version);
+
+			// Try to fish a scope out of the parent POM
+			if (string.IsNullOrEmpty (dependency.Scope))
+				dependency.Scope = parent_dependency?.Scope;
+		}
+
+		var version = dependency.Version;
+
+		if (string.IsNullOrWhiteSpace (version))
+			return;
+
+		version = ReplaceVersionProperties (project, version);
+
+		// VersionRange.Parse cannot handle single number versions that we sometimes see in Maven, like "1".
+		// Fix them to be "1.0".
+		// https://github.com/NuGet/Home/issues/10342
+		if (version != null && !version.Contains ("."))
+			version += ".0";
+
+		dependency.Version = version;
+	}
+
+	static string? ReplaceVersionProperties (Project project, string? version)
+	{
+		// Handle versions with Properties, like:
+		// <properties>
+		//   <java.version>1.8</java.version>
+		//   <gson.version>2.8.6</gson.version>
+		// </properties>
+		// <dependencies>
+		//   <dependency>
+		//     <groupId>com.google.code.gson</groupId>
+		//     <artifactId>gson</artifactId>
+		//     <version>${gson.version}</version>
+		//   </dependency>
+		// </dependencies>
+		if (string.IsNullOrWhiteSpace (version) || project?.Properties == null)
+			return version;
+
+		foreach (var prop in project.Properties.Any)
+			version = version?.Replace ($"${{{prop.Name.LocalName}}}", prop.Value);
+
+		return version;
+	}
+
+	public static bool IsCompileDependency (this Dependency dependency) => string.IsNullOrWhiteSpace (dependency.Scope) || dependency.Scope.ToLowerInvariant ().Equals ("compile");
+
+	public static bool IsRuntimeDependency (this Dependency dependency) => dependency?.Scope != null && dependency.Scope.ToLowerInvariant ().Equals ("runtime");
```

Comment:
For these can you use instead:

```suggestion
	public static bool IsCompileDependency (this Dependency dependency) => string.IsNullOrWhiteSpace (dependency.Scope) || string.IndexOf("compile", StringComparison.OrdinalIgnoreCase) != -1;

	public static bool IsRuntimeDependency (this Dependency dependency) => dependency?.Scope != null && string.IndexOf("runtime", StringComparison.OrdinalIgnoreCase) != -1;
```

To avoid creating strings. There is an overload for `string.Equals()`, too, but I don't think it exists in netstandard2.0.